sql/solutions/lesson6.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: page
+title: Lesson 6 solutions to exercises
+---
+# Lesson 6 solutions to exercises
+
+## L6.1 List all the hosts that had at least one workshop hosted.
+
+```SQL
+SELECT h.name
+  FROM hosts h
+ WHERE EXISTS (
+          SELECT *
+            FROM workshops w
+           WHERE h.id = w.host_id
+         );
+```
+## L6.2 List all the persons that have someone else with their same age (Use only JOINs)
+
+```SQL
+SELECT DISTINCT p1.name, p1.age
+  FROM people p1
+       INNER join people p2
+       ON p1.age = p2.age
+ WHERE p1.name <> p2.name;
+```
+## L6.3 List all the persons that have someone else with their same age (Use only EXISTS)
+
+```SQL
+SELECT p1.name
+  FROM people p1
+ WHERE EXISTS (
+         SELECT p2.name
+           FROM People p2
+          WHERE p1.age = p2.age
+                AND p1.name <> p2.name
+       );
+
+```
+## L6.4 List all the pair of persons that met at least in one workshop.
+
+```SQL
+SELECT p1.name, p2.name, r2.workshop_id
+  FROM RSVP r1
+       INNER JOIN people p1
+       ON r1.person_id = p1.id
+       INNER JOIN rsvp r2
+       ON r1.workshop_id = r2.workshop_id
+       INNER JOIN people p2
+       ON r2.person_id = p2.id
+ WHERE r1.Attendance = 1
+       AND r2.Attendance = 1
+       AND p1.name <> p2.name;
+
+```
+## L6.5 List all the persons with the number of workshops each one has been to.
+```SQL
+SELECT p.name, COUNT (*)
+  FROM people p
+       INNER JOIN rsvp r
+       ON p.id = r.person_id
+ WHERE attendance = 1
+ GROUP BY p.name;
```

Comment:
Same here, specially for grouping, as there could be 2 persons with the same name, best is to group by p.id to be sure that you only group records of the same person